Rationale.md

```diff
@@ -0,0 +1,215 @@
+# Design Rationale
+
+WebAssembly was designed incrementally, and desisions were made with a solid
+rationale which may turn out to be incorrect in the future. Sometimes, decisions
+were made without the luxury of data, and the designers would like to revisit
+certain decisions once real end-to-end usecases are met and Science can be
+performed.
+
+You, the reader, may also be interested in understanding why
+[Abstract Syntax Tree (AST)](AstSemantics.md) semantics are the way they are,
+and digging through github's issues and pull requests becomes difficult.
```

Comment:
I think it's useful :-)

Which part don't you agree with? I'm open to changing it, but I do think the text makes stuff easier to understand.